packages/app-folders/src/handlers.tsx

```diff
@@ -10,40 +10,33 @@ import { Loading } from "~/types";
  * @param apolloQuery: Apollo Query or Mutation
  */
 export const apolloFetchingHandler = async (
-    loadingHandler: () => void,
+    loadingHandler: (flag: boolean) => void,
     apolloQuery: () => Promise<ApolloQueryResult<any> | FetchResult<any>>
 ) => {
-    loadingHandler();
+    loadingHandler(true);
 
     const response = await apolloQuery();
-
-    loadingHandler();
+    loadingHandler(false);
 
     return response;
 };
 
 /**
  * A simple function to handle `loading` state.
  *
- * @param context: string that define where the context the operation works.
  * @param action: the `action` that has been performed.
  * @param setState: the logic to update the loading state.
  */
 export const loadingHandler = <T extends string>(
-    context: string,
     action: T,
     setState: Dispatch<SetStateAction<Loading<T>>>
-): void => {
-    setState(state => {
-        const currentContext = state[context] || {};
-        const currentAction = currentContext[action] || false;
-
-        return {
-            ...state,
-            [context]: {
-                ...currentContext,
-                [action]: !currentAction
-            }
-        };
-    });
+) => {
```

Comment:
Also add use `LoadingActions` as the `action` type here. That way you have autocomplete, and type safety.